app/models/conversation.rb

```diff
@@ -31,6 +31,10 @@ def hiring_fee_eligible?
     developer_replied? && created_at <= 2.weeks.ago
   end
 
+  def self.first_message?(developer)
+    where(developer:).count == 1 && where(developer:).first.messages.count == 1
+  end
```

Comment:
Let's move this to `Message` and use the following query since it will only make a single database call.

```ruby
class Message < ApplicationRecord
  def self.first_message?(developer)
    joins(:conversation).where(conversation: {developer:}).one?
  end
end
```

Reply:
Done, I moved those method to `Message` and modify the query